benchmark/inference/inference_benchmark.py

```diff
@@ -0,0 +1,126 @@
+
+                for layers in args.num_layers:
+                    if hetero:
+                        subgraph_loader = NeighborLoader(
+                            copy.copy(data),
+                            num_neighbors=[args.hetero_num_neighbors] * layers,
+                            input_nodes=mask,
+                            batch_size=batch_size,
+                            shuffle=False,
+                            num_workers=args.num_workers,
+                        )
+                        subgraph_loader.data.n_id = torch.arange(
+                            data.num_nodes)
+
+                    for hidden_channels in args.num_hidden_channels:
+                        print(
+                            '-----------------------------------------------')
+                        print(f'Batch size={batch_size}, '
+                              f'Layers amount={layers}, '
+                              f'Hidden features size={hidden_channels}')
+                        params = {
+                            'inputs_channels': inputs_channels,
+                            'hidden_channels': hidden_channels,
+                            'output_channels': dataset.num_classes,
+                            'num_heads': args.num_heads,
+                            'num_layers': layers,
+                        }
+
+                        if model_name == 'pna_conv':
+                            if degree is None:
+                                degree = PNAConv.get_degree_histogram(
+                                    subgraph_loader)
+                                print(f'Calculated degree for {dataset_name}.')
+                            params['degree'] = degree
+
+                        model = get_model(
+                            model_name, params,
+                            metadata=data.metadata() if hetero else None)
+                        model = model.to(device)
+                        model.training = False
+
+                        start = default_timer()
+                        model.inference(subgraph_loader, device, progress_bar)
+                        stop = default_timer()
+                        print(f'Inference time={stop-start:.3f}\n')
+
+
+if __name__ == '__main__':
+    argparser = argparse.ArgumentParser('GNN inference benchmark')
+    argparser.add_argument('--datasets', nargs='+',
+                           default=['ogbn-mag', 'ogbn-products',
+                                    'reddit'], type=str)
+    argparser.add_argument(
+        '--models', nargs='+',
+        default=['edge_conv', 'gat', 'gcn', 'pna_conv', 'rgat',
+                 'rgcn'], type=str)
+    argparser.add_argument('--root', default='../../data', type=str)
+    argparser.add_argument('--eval-batch-sizes', nargs='+',
+                           default=[512, 1024, 2048, 4096, 8192], type=int)
+    argparser.add_argument('--num-layers', nargs='+', default=[2, 3], type=int)
```

Comment:
Same as comments above, let's unify this (along with hetero num neighbors) for both homogeneous and heterogeneous graphs.